desktop/apps/auctions2/templates/index.jsx

```diff
@@ -0,0 +1,35 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import { withLayout } from 'desktop/components/react/main_layout'
+
+function Index (props) {
+  const {
+    Layout: { Header, Body, Footer },
+    Templates: { Meta }
+  } = props
+
+  return (
+    <div>
+      <Header>
+        <Meta />
+      </Header>
+      <Body>
+        hello!
+      </Body>
+      <Footer />
+    </div>
```

Comment:
I still wonder what we're really expressing here. It doesn't seem like we'll ever really want to rearrange the order of the `Layout`'s components. So, why not make that framing implicit? Something like:

```
import PropTypes from 'prop-types'
import React from 'react'
import { withLayout } from 'desktop/components/react/main_layout'

function Index () {
  return (
    <div>hello!</div>
  )
}

export default withLayout(Index)
```

So `withLayout` would effectively inject its component into the body block. Or I guess it would probably be less of an `HOC` than a renderer, since the outcome would be a fully rendered layout, instead of a component. I confirmed locally that it's possible to put raw HTML in a Pug `block`.

Then, to the extent that we need to be able to hook into those other sections, perhaps that could be done as additional arguments to `withLayout`. So you'd have `withLayout(Index, { header })` if you needed to add something to the header.

Or, if we need a lot of control over ways to manipulate the subsections, `withLayout` could become basically a constructor for a [fluent interface](https://en.wikipedia.org/wiki/Fluent_interface#JavaScript) that offers the ability to attach or overwrite the different sections, with either React components or raw HTML. So you could have operations like `replaceFooterWithComponent`, `appendRawHtmlToHeader`, or `appendMeta`, and terminate with a `renderToString()`.